### PR TITLE
Feature clean architecture

### PR DESCRIPTION
**Pull request by trishaagaba**
**What does this PR do**
-Implements the clean architecture in the code
-Adds the Splash screen
-Implements the filter widget

**Things to do**
-Introduce pagination to my list
-Document the ReadMe

### CODE REVIEW

File: lib/features/user_lists/presentation/providers/user_provider.dart
Comment (NabunyaLilian) [inline lib/features/user_lists/presentation/providers/user_provider.dart:1]: At this point you shouldn't be communicating with the Datasource. This is the point where usecases come in

File: lib/features/user_lists/presentation/screens/home.dart
Comment (NabunyaLilian) [inline lib/features/user_lists/presentation/screens/home.dart:1]: This is good work, but i suggest you break the widget into smaller widgets . let the file not exceed 200 lines of code

File: lib/features/user_profile/presentation/screens/user_profile_page.dart
Comment (NabunyaLilian) [inline lib/features/user_profile/presentation/screens/user_profile_page.dart:1]: fetchUserDetails method should not be in the widget

File: lib/features/user_lists/presentation/providers/user_provider.dart
Comment (trishaagaba) [reply lib/features/user_lists/presentation/providers/user_provider.dart:1]: alright let me look into it. Thank you

File: lib/features/user_profile/presentation/screens/user_profile_page.dart
Comment (trishaagaba) [reply lib/features/user_profile/presentation/screens/user_profile_page.dart:1]: Alright let me clean up that feature

File: lib/features/user_lists/presentation/screens/home.dart
Comment (trishaagaba) [reply lib/features/user_lists/presentation/screens/home.dart:1]: Alright thank you. Let me make the necessary adjustments

File: lib/features/user_lists/data/datasources/remote/data_source.dart
Comment (NabunyaLilian) [inline lib/features/user_lists/data/datasources/remote/data_source.dart:1]: Theres's some confusion here regarding the url, please clear it up. Also at this point we deal with models not entities.

File: lib/features/user_lists/data/repository/user_repository_impl.dart
Comment (NabunyaLilian) [inline lib/features/user_lists/data/repository/user_repository_impl.dart:1]: This is the point where you do the conversion from a model object to an entity

File: lib/features/user_lists/presentation/providers/user_provider.dart
Comment (NabunyaLilian) [inline lib/features/user_lists/presentation/providers/user_provider.dart:1]: Look into this  "if (_isLoading) return;", what's it's significance

File: lib/features/user_profile/data/datasources/details_source.dart
Comment (NabunyaLilian) [inline lib/features/user_profile/data/datasources/details_source.dart:1]: At this point , we don't interact with entities, it's supposed to be models.

File: lib/features/user_profile/data/repository/userprofile_rep_imp.dart
Comment (NabunyaLilian) [inline lib/features/user_profile/data/repository/userprofile_rep_imp.dart:1]: Use class names that are easy to understand

File: lib/features/user_profile/presentation/widget/buildContent.dart
Comment (NabunyaLilian) [inline lib/features/user_profile/presentation/widget/buildContent.dart:1]: you could extract this into a helper function file "_launchURL" and reuse it

File: lib/features/user_profile/presentation/widget/buildTop.dart
Comment (NabunyaLilian) [inline lib/features/user_profile/presentation/widget/buildTop.dart:1]: what is "topDist", use names that are easy to understand

**Review by NabunyaLilian — CHANGES_REQUESTED** [review]
Good work 👍 Trisha , just make a few adjustments on the changes i've requested and then we are good to go.

**Comment by trishaagaba** [comment]
Ready for merge

**Review by NabunyaLilian — APPROVED** [review]
LGTM, good work Trisha

**Pull request merged**